exonum/src/runtime/execution_context.rs

```diff
@@ -24,11 +24,17 @@ use crate::{
     },
 };
 
+const ACCESS_ERROR_STR: &str = "An attempt to access blockchain data after execution error.";
+
 /// Provides the current state of the blockchain and the caller information for the call
 /// which is being executed.
 ///
 /// The call can mean a transaction call, `before_transactions` / `after_transactions` hook,
 /// or the service constructor invocation.
+///
+/// # Safety
+///
+/// Errors that occur after making nested calls must bubble up to the upper level.
```

Comment:
Nit: I'd add the following sentence:

> If an error has occurred in a nested call, but the returned result of the topmost call is `Ok(())`, the latter will be coerced to an error (`CoreError::InvalidCall`) and recorded as such in the blockchain. Accessing storage after an error in a nested call will result in a panic.

I'd also change "must bubble up" to "should be bubbled up" since the execution safety *does not* depend on whether the service logic is implemented according to this rule.

Finally, you should mention that making child calls is unstable functionality. Perhaps, it could make sense to move the docs to `ExecutionContextUnstable` because of this.